homeassistant/components/nobo_hub/select.py

```diff
@@ -0,0 +1,169 @@
+    """Global override selector for Nobø Ecohub."""
+
+    _attr_has_entity_name = True
+    _attr_translation_key = "global_override"
+    _attr_device_class = "nobo_hub__override"
+    _attr_should_poll = False
+    _modes = {
+        nobo.API.OVERRIDE_MODE_NORMAL: "none",
+        nobo.API.OVERRIDE_MODE_AWAY: "away",
+        nobo.API.OVERRIDE_MODE_COMFORT: "comfort",
+        nobo.API.OVERRIDE_MODE_ECO: "eco",
+    }
+    _attr_options = list(_modes.values())
+    _attr_current_option: str
+
+    def __init__(self, hub: nobo, override_type) -> None:
+        """Initialize the global override selector."""
+        self._nobo = hub
+        self._attr_unique_id = f"{hub.hub_serial}"
+        self._override_type = override_type
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, f"{hub.hub_serial}")},
+            name=hub.hub_info[ATTR_NAME],
+            manufacturer=NOBO_MANUFACTURER,
+            model=f"Nobø Ecohub ({hub.hub_info[ATTR_HARDWARE_VERSION]})",
+            sw_version=hub.hub_info[ATTR_SOFTWARE_VERSION],
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Register callback from hub."""
+        self._nobo.register_callback(self._after_update)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Deregister callback from hub."""
+        self._nobo.deregister_callback(self._after_update)
+
+    async def async_select_option(self, option: str) -> None:
+        """Set override."""
+        mode = [k for k, v in self._modes.items() if v == option][0]
+        try:
+            await self._nobo.async_create_override(
+                mode, self._override_type, nobo.API.OVERRIDE_TARGET_GLOBAL
+            )
+        except Exception as exp:
+            raise HomeAssistantError from exp
+
+    async def async_update(self) -> None:
+        """Fetch new state data for this zone."""
+        self._read_state()
+
+    @callback
+    def _read_state(self) -> None:
+        for override in self._nobo.overrides.values():
+            if override["target_type"] == nobo.API.OVERRIDE_TARGET_GLOBAL:
+                self._attr_current_option = self._modes[override["mode"]]
+
+    @callback
+    def _after_update(self, hub) -> None:
+        self._read_state()
+        self.async_write_ha_state()
+
+
+class NoboProfileSelector(SelectEntity):
+    """Week profile selector for Nobø zones."""
+
+    _attr_translation_key = "week_profile"
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+    _profiles: dict[int, str] = {}
+    _attr_options: list[str] = []
+    _attr_current_option: str
+
+    def __init__(self, zone_id: str, hub: nobo) -> None:
+        """Initialize the week profile selector."""
+        self._id = zone_id
+        self._nobo = hub
+        self._attr_unique_id = f"{hub.hub_serial}:{zone_id}:profile"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, f"{hub.hub_serial}:{zone_id}")},
```

Comment:
Do we really need a separate device for each zone, shouldn't these be included in the hub device?

Reply:
This is how I was adviced to do it for the climate entity. Each zone can consist of one or more heaters or floor thermostats, so the actual physical devices are not included. This identifier matches the device info in the climate entity.

(I've since discovered that some floor thermostats requires a separate zone. A possibility is to use the actual device in device info when a zone only consists of one device, but this would require a similar change in the climate entity. We would also need to decide what device (or none) to use for zones that consists of multiple devices.)

Reply:
Ah, I didn't realize this device id is already used in the climate entity, in that case it's fine 👍